tools/misc/tpm2_encodeobject.c

```diff
@@ -186,6 +190,16 @@ static int encode(void) {
     }
 
     PEM_write_bio_TSSPRIVKEY_OBJ(bio, tpk);
+
+    if ((ctx.autoflush || tpm2_util_env_yes(TPM2TOOLS_ENV_AUTOFLUSH)) &&
+        ctx_file_exists(ctx.parent.ctx_path) &&
```

Comment:
@JuergenReppSIT no need to check `ctx_file_exists(ctx.parent.path)` here, because you can check `ctx.parent.object.path`. If that field is set than you know for a fact it is a context file and it existed when the tool was opened and avoids the race condition of the check and the filesystem. Suppose the tool has the context file unlinked from out and underneath it, we still want to flush the transient handle.

Reply:
@williamcroberts Sorry for being a bit confused here ;-) You are right, the test of the path in  `tpm2_loaded_object` is sufficient.